stdlib/src/memory/maybe_uninitialized.mojo

```diff
@@ -250,4 +267,12 @@ struct UnsafeMaybeUninitialized[ElementType: AnyType](CollectionElementNew):
         If the memory was initialized, the caller should
         use `assume_initialized_destroy` before.
         """
-        pass
+
+        @parameter
+        @always_inline
+        fn _drop[T: AnyType](owned val: T):
```

Comment:
**Question** What's the value in wrapping this MLIR op call in an inner function instead of just inlining it?

Reply:
Ah, I see now from your comment in the PR description - can you check if this is required still?  @lattner has improved this subobject destruction area recently in `CheckLifetimes`, so it may not be needed anymore.

Reply:
I'll add this to my todo list tomorrow.

For some reason i cant comment on your review below, but the reason the move count goes up by one is because I'm doing two moves instead of one now. The actual moves should get optimized away, but it cant optimize the counter increment because that would be incorrect. (i'll explore cleaning it up some more though and getting rid of the extra move)

Reply:
moving it out of the closure still complains about not being able to destroy subobjects, but maybe there's a better way

Reply:
but i can mark subobjects initialized now, which helps

Reply:
i'm cooking